peril.settings.json

```diff
@@ -18,14 +18,17 @@
     "artsy/reaction": {
       "pull_request": "danger/pr.ts"
     },
+    "artsy/force": {
+      "pull_request": "danger.ts"
```

Comment:
out of curiosity (and i suspect not) but can Peril interop with danger ruby? Like would `pull_request: Dangerfile` be valid?

Reply:
GReeeeeaaat question.

The simplest answer is no, today... and probably tomorrow.

Our Peril is ["Peril Staging"](https://github.com/danger/peril/blob/master/docs/using_peril_staging.md) - which separates Peril the Webhooks API system, from Peril. the Danger runtime. The Danger runtime is a docker instance that runs a single CLI command with JSON that Peril sends with the whole environment. This effectively DIs in everything Danger needs to make a run (auth, git changes, Webhooks metadata etc) see an [example json](https://github.com/danger/peril/blob/master/source/github/events/handlers/_tests/fixtures/PerilRunnerPRBootStrapExample.json).

Danger runs as a local process in the docker container, which actually is the same architectural system that Danger JS runs locally. The main problem is that you can't know when async work has finished without knowing when the process is over, so danger runs your code in a subprocess. *This abstraction is powerful*. It means the process doesn't have to be a nodeJS process. 

So if you read [Danger in your Language](https://danger.systems/js/usage/danger-process.html) (the official docs for `danger process`) you can see a 10 liner implementation of Danger in Ruby

```ruby
  #!/usr/bin/env ruby

require 'json'
dsl_json = STDIN.tty? ? 'Cannot read from STDIN' : $stdin.read
danger = JSON.parse(dsl_json)
results = { warnings: [], messages:[], fails: [], markdowns: [] }

if danger.github.pr.body.include? "Hello world"
  results.messages << { message: "Hey there" }
end

require 'json'
STDOUT.write(results.to_json)
```

I could make a docker container that use ^ as a runtime, and then a lot of Danger Ruby could be ported.

That said "Danger Ruby" the official one, has very different architecture that something like the above because it aims to build a pretty custom DSL with plugins etc - that has all of the ruby magic. So it's not directly portable.

Reply:
Ahhh thanks for the explaination, makes sense 👍